desktop/cmp/dash/canvas/DashCanvasModel.js

```diff
@@ -320,23 +343,27 @@ export class DashCanvasModel extends HoistModel {
                 title: title ?? viewSpec.title,
                 containerModel: this
             }),
-            prevLayout = previousViewId ? this.getLayout(previousViewId) : null,
+            prevLayout = previousViewId ? this.getViewLayout(previousViewId) : null,
             x = prevLayout?.x ?? layout?.x ?? 0,
             y = prevLayout?.y ?? layout?.y ?? this.rows,
             h = layout?.h ?? viewSpec.height ?? 1,
             w = layout?.w ?? viewSpec.width ?? 1;
 
-        this.layout = [...this.layout, {i: id, x, y, h, w}];
+        this.setLayout([...this.layout, {i: id, x, y, h, w}]);
         this.viewModels = [...this.viewModels, model];
         return model;
     }
 
     @action
     setLayout(layout) {
-        // strip extra properties from react-grid
-        layout = layout.map(({i, x, y, w, h}) => ({i, x, y, w, h}));
-        if (!isEqual(this.layout, layout)) {
+        // strip extra properties from react-grid-layout
+        layout = layout.map(it => pick(it, ['i', 'x', 'y', 'w', 'h']));
+
+        const layoutChanged = !isEqual(sortBy(this.layout, 'i'), sortBy(layout, 'i'));
+
+        if (layoutChanged) {
```

Comment:
Could change this to an early out to avoid the extra nesting `if (!this.layoutChanged) return;`